File: frontend_analisis_exergetico/src/modules/exergia/models/proyecto.ts
import { BaseModel } from 'src/utils/types';
import { Flujo } from './flujo';

export interface ProyectoBase extends BaseModel {
  nombre: string;
  descripcion?: string;
}

export type ProyectoCreateBase = ProyectoBase;
export type ProyectoUpdateBase = ProyectoBase;

export interface Proyecto extends ProyectoBase {
  id: number;
  idFlujoEntrada: number;
  idFlujoSalida: number;
  flujoEntrada: Flujo;
  flujoSalida: Flujo;
}

File: backend_analisis_exergetico/schemas/proyecto.py
from fastapi_utils.api_model import APIModel

from schemas.flujo import Flujo


class ProyectoBase(APIModel):
    nombre: str
    descripcion: str
    id_flujo_entrada: int | None
    id_flujo_salida: int | None


class ProyectoCreate(ProyectoBase):
    pass


class ProyectoUpdate(ProyectoBase):
    pass


class Proyecto(ProyectoBase):
    id: int
    flujo_entrada: Flujo | None
    flujo_salida: Flujo | None
